kubernetes/src/main/scala/org/apache/spark/deploy/kubernetes/SparkJobResource.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.apache.spark.deploy.kubernetes
+
+import java.util.concurrent.Executors
+
+import io.fabric8.kubernetes.client.{BaseClient, KubernetesClient}
+import okhttp3.{MediaType, OkHttpClient, Request, RequestBody}
+import org.apache.spark.SparkException
+import org.apache.spark.internal.Logging
+import org.json4s._
+import org.json4s.jackson.JsonMethods._
+import org.json4s.JsonDSL._
+
+class SparkJobResource(client: KubernetesClient) extends Logging {
+  val ApiVersion = "apache.io/v1"
+  val Kind = "SparkJob"
+
+//  apiVersion: "apache.io/v1"
+//  kind: "SparkJob"
+//  metadata:
+//    name: "spark-job-1"
+//  spec:
+//    image: "driver-image"
+//    state: "completed"
+//    num-executors: 10
+
+  val httpClient = this.getHTTPClient(client.asInstanceOf[BaseClient])
+  def createJobObject(name: String, executors: Int, image: String): Unit = {
+    val json = ("apiVersion" -> ApiVersion) ~
```

Comment:
Using the case classes definitely makes sense. If you have sorted out the CLA/licence issues, you could merge in the contents of your work as a commit on this PR. Otherwise, if you feel that your change is more comprehensive, you could put up your PR and I can merge my commits into it.